.circleci/config.yml

```diff
@@ -43,27 +49,70 @@ jobs:
           name: "Install Rust Build Toolchain"
           command: |
             curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > /tmp/rustup-init
-            sh /tmp/rustup-init -y --default-toolchain stable
-            . "$HOME"/.cargo/env
+            for i in $(seq 10); do
+                if sh /tmp/rustup-init -y --default-toolchain stable; then
+                    break
+                else
+                    sleep 1
+                fi
+            done
+            echo '. "$HOME"/.cargo/env' >> $BASH_ENV
 
       - run:
           name: "Get Python Build/Package Dependencies"
           command: |
             # Make sure we have a pip that's aware of Python version constraints
-            pip install --upgrade pip
+            << parameters.python >> -m pip install --upgrade pip
             # Pre-install these two setup_depends using pip so we don't have
             # to rely on whatever resolution logic setuptools would apply.
-            pip install --upgrade milksnake setuptools_scm
-            # And get these so we can build and then upload a wheel
-            pip install wheel twine
+            << parameters.python >> -m pip install --upgrade milksnake setuptools_scm
+            # And get these so we can build, fix, and upload a wheel.
+            << parameters.python >> -m pip install wheel twine auditwheel
 
       - run:
           name: "Build Wheel"
           command: |
-            python setup.py bdist_wheel
+            << parameters.python >> -m pip wheel .
```

Comment:
I'd suggest using `python -m build` (which uses [build](https://pypi.org/project/build/)) or at least passing `--no-deps`. The current command builds (or fetches) wheels for all dependencies of the specified packages as well.

It looks like build puts the files in `dist/`, instead of the current directory.

Reply:
Hm.  It looks like `python -m build` has a problem with the way the package is set up:

```
* Building wheel from sdist                                                                                                                                                       
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools >= 40.8.0, wheel)        
* Getting dependencies for wheel... 
* Installing packages in isolated environment... (milksnake, setuptools_scm, wheel)   
* Building wheel...                 
running bdist_wheel                                                                      
running build                                                                            
running build_py                                                                         
creating build/lib                                                                       
creating build/lib/challenge_bypass_ristretto                                        
copying challenge_bypass_ristretto/__init__.py -> build/lib/challenge_bypass_ristretto
creating build/lib/challenge_bypass_ristretto/tests
copying challenge_bypass_ristretto/tests/__init__.py -> build/lib/challenge_bypass_ristretto/tests
copying challenge_bypass_ristretto/tests/test_privacypass.py -> build/lib/challenge_bypass_ristretto/tests
error: [Errno 2] No such file or directory: './challenge-bypass-ristretto-ffi'

ERROR Backend subproccess exited when trying to invoke build_wheel
```

If I take the progress indicator "Building wheel from sdist" at face value then I suppose the problem is that challenge-bypass-ristretto does not get included in our sdist so isn't available when this stage is reached.

Maybe addressing this and switching to `python -m build` would be a reasonable follow-up?  Since it appears to be using `bdist_wheel` under the hood, just like `pip wheel` does, perhaps the wheels themselves are the same either way and `python -m build` is only preferable for project maintenance reasons?  (In which case a follow-up fix seems fine)

Adding `--no-deps` seems trouble-free though so that much I can do now.

Reply:
I went to file a ticket for using `python -m build` and I realized I couldn't quite articulate the reason for preferring it.  https://pypi.org/project/build/ says it is "a .. package builder" but doesn't try to claim to be the definitive package builder or anything like that.  I'm not sure what the pros and cons of it vs `pip wheel` are.  Do you feel like filing this ticket?

Reply:
There are a couple of things:
- partly it is that other tools are not the right choice:
  - directly invoking `setup.py` is deprecated (see for example [this comment](https://github.com/pypa/setuptools/issues/2088#issuecomment-623567153) and [this section](https://setuptools.pypa.io/en/latest/build_meta.html#what-is-it) of setuptools documentation) 
  - `pip wheel` is intended to get wheels of all dependencies (possibly fetching them from pypi, or building them from sdists)
- as for why build, it is the suggestion in setuptools' [quickstart](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use) and by pep517's [readme](https://github.com/pypa/pep517#deprecated-high-level) (the pep517 package  is the implementation of the new API for building packages which is used by pip).

> If I take the progress indicator "Building wheel from sdist" at face value then I suppose the problem is that challenge-bypass-ristretto does not get included in our sdist so isn't available when this stage is reached.

Yeah, it seems that (by default), `build` builds a sdist, and then builds a wheel from that. That suggests that the sdist that is produced can't actually be installed, which would be a reasonable thing to fix. You can pass `-w` to `build`, to get it to just build a wheel directly from source, which works.